### PR TITLE
Fixed issue #146: DOS MZ Exe misidentified as NE Exe

### PR DESCRIPTION
**Pull request by powerbf**
Look for NE header instead of relying on reloc table offset which can throw up a false positive

### CODE REVIEW

**Comment by ceeac** [comment]
I think it is better to directly reject NE executables in `ExeBinaryLoader::canLoad`, to give other plugins a chance to read the file. I will do a more detailed review tomorrow.

**Comment by powerbf** [comment]
Agreed. This check should be in canLoad(). I've moved it.

File: src/boomerang-loaders/exe/dos/ExeBinaryLoader.cpp [hunk 1]
```diff
@@ -260,12 +255,33 @@ int ExeBinaryLoader::canLoad(QIODevice &fl) const
...
+
+    /* Check if this is a  Windows NE executable
+     * It will have a DOS MZ header, but also another header later on
+     * Offset of the NE header is at 0x3C
+     * The NE header will start with the signature 'NE'
+     * Ref: https://www.fileformat.info/format/exe/corion-mz.htm
+     */
+    if (fl.seek(0x3C)) {
+        if (fl.read(reinterpret_cast<char *>(buf), 2) == 2) {
+            SWord possibleWinHeaderOffset = Util::readWord((void *)buf, Endian::Little);
```
[the remark this answers is not in the document]
Comment (powerbf) [reply src/boomerang-loaders/exe/dos/ExeBinaryLoader.cpp]: done

**Comment by powerbf** [comment]
Replaced by pull request #150

**Pull request closed without merging**